components/sidebar/sidebar.component.js

```diff
@@ -41,52 +60,71 @@ class Sidebar extends Component {
 
     // Get Props
     static getDerivedStateFromProps(props,state) {
-
         return {
             toc: Array.isArray(props.toc) ? props.toc : []
         }   
     }
 
     // Render TOC
-    toc(routes, depth) {
+    loop = (routes) => {
 
         // Array to store jsx of routes
-        var processed = [];
+        var processed= [];
+        var pinned =[];
 
         // Loop over route
         for (var route of routes) {
 
             // If it is a file i.e. root node
             // Then simply convert it into right format
-            if (route.meta) processed.push(<Route key={cuid()} title={route.meta.title} depth={depth} path={route.path} />)
-            else {
-                // If it is a directory, then we gotta do another
-                // pass to also tranform all the children
-                processed.push(<div key={cuid()} className={`mt-5 mb-1 text-white-2 text-base ${ depth >= 0 ? "" : "uppercase" }`} style={{ marginLeft: `${ depth <= 0 ? 0 : depth * 10 }px` }}>{route.key}</div>)
-                processed.push(...this.toc(route.children, depth + 1));
+            if (route.meta){
+                // If File is Pinned
+                if (route.meta.pinned) {
+                    pinned.push(<Route key={cuid()} title={route.meta.title} path={route.path} emoji={route.meta.emoji} ></Route>) 
+                }
+                // If File is Not Pinned
+                else {
+                    processed.push(<Route key={cuid()} title={route.meta.title} path={route.path}></Route>)
+                }
             }
 
-        }
+            // If it is a directory, then we gotta do another
+                // pass to also tranform all the children
+            else{
```

Comment:
Space before brace. I recommend you read https://google.github.io/styleguide/jsguide.html